File: 128-longest-consecutive-sequence/main.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        
        nums_set = set(nums)
        max_length: int = 0

        for n in nums_set:
            if (n - 1) not in nums_set:
                length = 1
                while (n + length) in nums_set:
                    length += 1
                max_length = max(length, max_length)
        
        return max_length

File: 128-longest-consecutive-sequence/main.rs
use std::collections::HashSet;

impl Solution {
    pub fn longest_consecutive(nums: Vec<i32>) -> i32 {

        let mut nums_set: HashSet<i32> = nums.into_iter().collect();
        let mut longest = 0;

        for &num in &nums_set {
            if !nums_set.contains(&(num - 1)) {
                let mut length = 1;
                while nums_set.contains(&(num + length)) {
                    length += 1;
                }

                longest = longest.max(length);
            }
        }

        longest
    }
}
